x/twap/store.go

```diff
@@ -147,7 +147,6 @@ func (k Keeper) getRecordAtOrBeforeTime(ctx sdk.Context, poolId uint64, t time.T
 		if lastParsedTime.After(parsedTime) {
 			return true
 		}
-		lastParsedTime = parsedTime
```

Comment:
This is needed, lastParsedTime is defined out of the scope

Reply:
I added it back, but still don't quite understand its purpose. It updates lastParsedTime which is defined out of scope, but as far as I see, lastParsedTime is not used anywhere else but in that scope